dotCMS/src/test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentAsDraftActionletTest.java

```diff
@@ -0,0 +1,149 @@
+import static org.mockito.Mockito.*;
+
+@RunWith(DataProviderRunner.class)
+public class SaveContentAsDraftActionletTest {
+
+    private SaveContentAsDraftActionlet saveContentAsDraftActionlet;
+    private ContentletAPI contentletAPI;
+    private CategoryAPI categoryAPI;
+    private PermissionAPI permissionAPI;
+
+    @Before
+    public void init(){
+        contentletAPI = mock(ContentletAPI.class);
+        categoryAPI = mock(CategoryAPI.class);
+        permissionAPI = mock(PermissionAPI.class);
+
+        saveContentAsDraftActionlet = new SaveContentAsDraftActionlet(contentletAPI, categoryAPI, permissionAPI);
+    }
+
+    private static class TestCase {
+        final private boolean respectFrontendRoles;
+
+        private TestCase(boolean respectFrontendRoles) {
+            this.respectFrontendRoles = respectFrontendRoles;
+        }
+    }
+
+    @DataProvider
+    public static Object[] respectFrontendRolesValues() {
+        return new TestCase[]{new TestCase(true), new TestCase(false)};
+    }
+
+    /**
+     * When: {@link WorkflowProcessor#getContentletDependencies()} is not null
+     * Should: call {@link ContentletAPI#saveDraft(Contentlet, Map, List, List, User, boolean)} with the right parameters
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    @UseDataProvider("respectFrontendRolesValues")
+    public void shouldCallSaveDraftWithTheRightParameters(final TestCase testCase) throws DotSecurityException, DotDataException {
+        final WorkflowProcessor processor = mock(WorkflowProcessor.class);
+        final Map<String, WorkflowActionClassParameter> params = map();
+
+        final Contentlet contentlet = mock(Contentlet.class);
+        final User user  = mock(User.class);
+
+        when(processor.getUser()).thenReturn(user);
+        when(processor.getContentlet()).thenReturn(contentlet);
+
+        List<Category> categories = mock(List.class);
+
+        final List<Permission> permissions = mock(List.class);
+        when(this.permissionAPI.getPermissions(contentlet, false, true)).thenReturn(permissions);
+
+        final ContentletDependencies contentletDependencies = mock(ContentletDependencies.class);
+        when(processor.getContentletDependencies()).thenReturn(contentletDependencies);
+
+        final Contentlet contentletNew = mock(Contentlet.class);
+        when(contentletDependencies.getCategories()).thenReturn(categories);
+
+        when(processor.getContentletDependencies().isRespectAnonymousPermissions()).thenReturn(testCase.respectFrontendRoles);
+
+        when(this.contentletAPI.saveDraft(
+                contentlet, (ContentletRelationships) null, categories, permissions, user, testCase.respectFrontendRoles))
+                .thenReturn(contentletNew);
+
+        saveContentAsDraftActionlet.executeAction(processor, params);
+
+        //verify(processor, times(1)).setContentlet(contentletNew);
```

Comment:
remove this commented line

Reply:
really this line need to be uncomment

https://github.com/dotCMS/core/pull/18300/commits/1aa3444ac01b694697c3580a424d7e3566aabc35#diff-4e93f1c0bcf3f04b2c6542297d6360eeR97